src/app/dataset-view/additional-components/dataset-settings-component/tabs/access/dataset-settings-access-tab/add-people-modal/add-people-modal.component.html

```diff
@@ -0,0 +1,118 @@
+<div class="d-flex p-4">
+    <div class="flex-grow-1 ms-2">
+        <span class="fs-5 fw-bold">{{ collaborator ? "Edit role" : "Add people to dataset" }}</span>
+    </div>
+    <div>
+        <button
+            type="button"
+            class="btn-close"
+            aria-label="Close"
+            (keydown.esc)="activeModal.close()"
+            (click)="activeModal.close()"
+        ></button>
+    </div>
+</div>
+<div class="px-4">
+    <mat-divider />
+</div>
+
+<div class="body p-4">
+    <div *ngIf="!collaborator && !selectedCollaborator" class="d-flex flex-column position-relative">
+        <label for="" class="form-label d-block">Search by account name, display info </label>
```

Comment:
This template is very large and has complicated conditions.
It feels like it will be easier to have separate components to:
 - add a person with search form and specify it's role
 - edit a predefined person and change it's role
 
 If you will have common markup, such as for role selection, you can always extract sub-component.

Reply:
Separate into 2 components